node/src/main/scala/org/bitcoins/node/PeerFinder.scala

```diff
@@ -0,0 +1,257 @@
+package org.bitcoins.node
+
+import akka.actor.{ActorSystem, Cancellable}
+import org.bitcoins.asyncutil.AsyncUtil
+import org.bitcoins.core.p2p.ServiceIdentifier
+import org.bitcoins.core.util.NetworkUtil
+import org.bitcoins.node.config.NodeAppConfig
+import org.bitcoins.node.models.{Peer, PeerDAO, PeerDb}
+
+import java.net.{InetAddress, UnknownHostException}
+import scala.collection.mutable
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.io.Source
+import scala.util.Random
+
+object PeerSource extends Enumeration {
```

Comment:
can we make this an ADT in Scala ? That way we can pattern match. 

IIUC, this would be

```scala
sealed trait PeerSource
object PeerSource {
  case object Config extends PeerSource
  ...
  case object Param extends PeerSource
}
```
you might want to `extend StringFactory` too so you can use `fromString()`

Reply:
Removed this for now as it's not used anywhere at the moment. Was doing some changes on top of this but they are not in this PR, will add when in another PR with those changes done.